platform/plugins/Places/web/js/tools/location.js

```
@@ -0,0 +1,326 @@
+(function (Q, $, window, document, undefined) {
+
+	/**
+	 * Places Tools
+	 * @module Places-tools
+	 */
+
+	var Users = Q.Users;
+	var Streams = Q.Streams;
+	var Places = Q.Places;
+
+	/**
+	 * Allows the logged-in user to add his own locations and select among these locations
+	 * @class Places location
+	 * @constructor
+	 * @param {Object} [options] used to pass options
+	 * @param {object} [options.geocode] Ready google location object
+	 * @param {bool} [options.useRelatedLocations] Whether to show related locations
+	 * @param {object} [options.location] Currently selected location
+	 * @param {Q.Event} [options.onChoose] this event occurs when user selected some valid location
+	 */
+
+	Q.Tool.define("Places/location", function (options) {
+			var tool = this;
+			var state = this.state;
+			var $te = $(tool.element);
+
+			if (!Users.loggedInUser) {
+				Users.login({
```

Instead of calling `Users.login`, please just throw an exception. The user of this tool is supposed to make sure that the user is already logged in. Do not allow the tool to be rendered otherwise.

Actually, you can show a useful interface to a user who is not logged-in also. Just don't show the "my saved locations" related tool, but you can show "My Current Location" button and a way to type in an address, why not. The user doesn't have to be logged in for it. Yes, please improve the tool to do that.